packages/confluence-adapter/src/filters/transform_template_body_to_template_expression.ts

```diff
@@ -0,0 +1,203 @@
+  spaceByKey: Record<string, InstanceElement>
+  pageBySpaceFullNameAndTitle: Record<string, Record<string, InstanceElement>>
+}
+
+// If you change one regex of a pair (TYPE_REF_REGEX, SPLIT_TYPE_REF_REGEX), you should change the other one as well
+const PAGE_REF_REGEX = /(<ri:page\s+ri:space-key="[^"]*"\s+ri:content-title="[^"]*"\s+ri:version-at-save="\d+"\s*\/>)/
+const SPLIT_PAGE_REF_REGEX =
+  /(<ri:page\s+ri:space-key=")([^"]*)("\s+ri:content-title=")([^"]*)("\s+ri:version-at-save="\d+"\s*\/>)/
+
+const SPACE_REF_REGEX = /(<ri:space\sri:space-key="[^"]*"\s*\/>)/
+const SPLIT_SPACE_REF_REGEX = /(<ri:space\sri:space-key=")([^"]*)("\s*\/>)/
+
+const handlePageRefMatch = (
+  matches: RegExpMatchArray,
+  indices: PossibleRefsInTemplateIndices,
+  fallback: string,
+): TemplatePart | TemplatePart[] => {
+  const { spaceByKey, pageBySpaceFullNameAndTitle } = indices
+  const [, spaceKey, spaceKeyValue, contentTitle, contentTitleValue, versionAtSave] = matches
+  const space = spaceByKey[spaceKeyValue]
+  if (space === undefined) {
+    log.warn('Could not find space with key %s', spaceKeyValue)
+    return fallback
+  }
+  const spaceReference = new ReferenceExpression(space.elemID, space)
+  const page = pageBySpaceFullNameAndTitle[space.elemID.getFullName()]?.[contentTitleValue]
+  if (page === undefined) {
+    log.warn(
+      'Could not find page with title %s in spaceKey %s, creating reference for space only',
+      contentTitleValue,
+      spaceKeyValue,
+    )
+    return [spaceKey, spaceReference, contentTitle, contentTitleValue, versionAtSave]
+  }
+  const pageReference = new ReferenceExpression(page.elemID, page)
+  return [spaceKey, spaceReference, contentTitle, pageReference, versionAtSave]
+}
+
+const handleSpaceRefMatch = (
+  matches: RegExpMatchArray,
+  spaceByKey: PossibleRefsInTemplateIndices['spaceByKey'],
+  fallback: string,
+): TemplatePart | TemplatePart[] => {
+  const [, spaceKey, spaceKeyValue, rest] = matches
+  const space = spaceByKey[spaceKeyValue]
+  if (space === undefined) {
+    log.warn('Could not find space with key %s', spaceKeyValue)
+    return fallback
+  }
+  const spaceReference = new ReferenceExpression(space.elemID, space)
+  return [spaceKey, spaceReference, rest]
+}
+
+const extractionFunc = (expression: string, indices: PossibleRefsInTemplateIndices): TemplatePart | TemplatePart[] => {
+  const pageMatches = expression.match(SPLIT_PAGE_REF_REGEX)
+  if (pageMatches !== null) {
+    return handlePageRefMatch(pageMatches, indices, expression)
+  }
+  const spaceMatches = expression.match(SPLIT_SPACE_REF_REGEX)
+  if (spaceMatches !== null) {
+    return handleSpaceRefMatch(spaceMatches, indices.spaceByKey, expression)
+  }
+  return expression
+}
+
+const prepRef = (ref: ReferenceExpression): TemplatePart => {
+  if (ref.value instanceof UnresolvedReference) {
+    log.trace(
+      'prepRef received a part as unresolved reference, returning an empty string, instance fullName: %s ',
```

Comment:
nits: 
1. I think this should be a debug log
2. we should also log the unresolved reference's elem id (there can be many)
3. 
```suggestion
      'prepRef received a part as unresolved reference, returning an empty string, instance fullName: %s',
```

Reply:
This code is from Zendesk, should I change it there as well?

Reply:
let's change only this one for now, I assume it was noisy in zendesk but let's see what happens here because it might indicate a bug (in zenedesk it's probably due to missing references which are unresolved in valid scenarios)